internal/servers/controller/handlers/authmethods/oidc.go

```diff
@@ -196,6 +201,54 @@ func (s Service) authenticateOidcStart(ctx context.Context, req *pbs.Authenticat
 	return &pbs.AuthenticateResponse{Command: req.GetCommand(), Attributes: attrs}, nil
 }
 
+func (s Service) authenticateOidcToken(ctx context.Context, req *pbs.AuthenticateRequest, authResults *auth.VerifyResults) (*pbs.AuthenticateResponse, error) {
+	const op = "authmethod_service.(Service).authenticateOidcToken"
+	if req == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "nil request")
+	}
+	if authResults == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "nil auth results")
+	}
+	if req.GetAttributes() == nil {
+		return nil, errors.New(errors.InvalidParameter, op, "nil request attributes")
+	}
+
+	attrs := new(pbs.OidcTokenAttributes)
+	if err := handlers.StructToProto(req.GetAttributes(), attrs); err != nil {
+		return nil, errors.New(errors.InvalidParameter, op, "error parsing request attributes")
+	}
+	if attrs.TokenId == "" {
+		return nil, errors.New(errors.InvalidParameter, op, "empty token id request attributes")
+	}
+
+	token, err := oidc.TokenRequest(ctx, s.kms, s.atRepoFn, attrs.TokenId)
+	if err != nil {
+		fmt.Println(err)
+		// TODO: Log something so we don't lose the error's context and entire msg...
+		switch {
+		case errors.Match(errors.T(errors.Forbidden), err):
+			return nil, errors.Wrap(err, op, errors.WithMsg("Forbidden"))
+		case errors.Match(errors.T(errors.AuthAttemptExpired), err):
+			return nil, errors.Wrap(err, op, errors.WithMsg("Forbidden"))
+		case errors.Match(errors.T(errors.InvalidParameter), err):
+			return nil, errors.New(errors.Internal, op, "Error invalid parameters for OIDC token request.")
+		default:
+			return nil, errors.New(errors.Internal, op, "Error during OIDC token request.")
+		}
+	}
+
+	attrsMap := map[string]interface{}{
```

Comment:
With password auth methods we can return the tokens as either a token on the response or as a split cookie.  This is determined based on the `token_type` value in the initial authentication request.

This is done by populating `token_type` in the response and letting internal/servers/controller/handlers/outgoing_interceptor.go intercept the response.

Reply:
Please let me know if I've done this correctly.

Reply:
@randallmorey Is it in the final `token` step of oidc authentication that FE is expected to provide the token type field?  If so, @jimlambrt , this looks right.  Thanks! :)

Reply:
Yes `token_type` is planned for the `token` step.